x-pack/plugins/alerting/server/plugin.ts

```diff
@@ -588,33 +591,36 @@ export class AlertingPlugin {
     };
 
     taskRunnerFactory.initialize({
-      logger,
+      actionsConfigMap: getActionsConfigMap(this.config.rules.run.actions),
+      actionsPlugin: plugins.actions,
+      alertsService: this.alertsService,
+      backfillClient: this.backfillClient!,
+      basePathService: core.http.basePath,
+      cancelAlertsOnRuleTimeout: this.config.cancelAlertsOnRuleTimeout,
+      connectorAdapterRegistry: this.connectorAdapterRegistry,
       data: plugins.data,
-      share: plugins.share,
       dataViews: plugins.dataViews,
-      savedObjects: core.savedObjects,
-      uiSettings: core.uiSettings,
       elasticsearch: core.elasticsearch,
-      getRulesClientWithRequest,
-      spaceIdToNamespace,
-      actionsPlugin: plugins.actions,
       encryptedSavedObjectsClient,
-      basePathService: core.http.basePath,
       eventLogger: this.eventLogger!,
       executionContext: core.executionContext,
-      ruleTypeRegistry: this.ruleTypeRegistry!,
-      alertsService: this.alertsService,
+      getMaintenanceWindowClientWithRequest,
+      getRulesClientWithRequest,
       kibanaBaseUrl: this.kibanaBaseUrl,
-      supportsEphemeralTasks: plugins.taskManager.supportsEphemeralTasks(),
-      maxEphemeralActionsPerRule: this.config.maxEphemeralActionsPerAlert,
-      cancelAlertsOnRuleTimeout: this.config.cancelAlertsOnRuleTimeout,
+      logger,
       maxAlerts: this.config.rules.run.alerts.max,
-      actionsConfigMap: getActionsConfigMap(this.config.rules.run.actions),
+      maxEphemeralActionsPerRule: this.config.maxEphemeralActionsPerAlert,
+      ruleTypeRegistry: this.ruleTypeRegistry!,
+      rulesSettingsService: new RulesSettingsService({
+        getRulesSettingsClientWithRequest,
+        isServerless: !!plugins.serverless,
+      }),
```

Comment:
actual changes are to pass this new global rules setting service to the task runner factory and to remove `getRulesSettingsClientWithRequest`